platform/ios/src/MGLMapView.mm

```diff
@@ -3995,20 +4000,21 @@ - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
             MGLAnnotationContext &annotationContext = _annotationContextsByAnnotationTag.at(annotationTag);
             annotationView = annotationContext.annotationView;
             if (annotationView && annotationView.enabled) {
-            {
-                // Annotations represented by views use the view frame as the positioning rect.
-                positioningRect = annotationView.frame;
-                [annotationView.superview bringSubviewToFront:annotationView];
-                [annotationView setSelected:YES animated:animated];
+                {
+                    // Annotations represented by views use the view frame as the positioning rect.
+                    positioningRect = annotationView.frame;
+                    [annotationView.superview bringSubviewToFront:annotationView];
+                    [annotationView setSelected:YES animated:animated];
+                }
             }
         }
-    }
 
-     // The client can request that any annotation be selected (even ones that are offscreen).
-     // The annotation can’t be selected if no part of it is hittable.
+    // The client can request that any annotation be selected (even ones that are offscreen).
+    // The annotation will bounce to the current tap point as anchor.
     if ( ! CGRectIntersectsRect(positioningRect, self.bounds) && annotation != self.userLocation)
     {
-        return;
+        CGPoint tapPoint = [singleTap locationInView:self];
```

Comment:
`singleTap` isn’t guaranteed to exist at this point. If the developer calls `-showAnnotation:animated:`, a public method, then `tapPoint` ends up being (0, 0). It should instead be the annotation’s center coordinate.